Add the block powered by MailPoet into template footer [MAILPOET-6416]

lysyjan opened the pull request:
## Description

This PR contains various fixes:
- Adding a block Power by MailPoet at the end of the template.
- Adding upsell modal to the block.
- Fix the dark logo.

## Code review notes

I chose the simplest solution for situations such as adding CSS styles to the block.

## QA notes

If the previous template has changed, it must be deleted from the database. It's a record in the wp_post table with the post_type `wp_template ` and the title `Newsletter`.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6416]

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-6416]: https://mailpoet.atlassian.net/browse/MAILPOET-6416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:email-editor/footer-logo)

_The latest successful build from `email-editor/footer-logo` will be used. If none is available, the link won't work._

lysyjan committed:
Add powered-by-mailpoet block to template footer

[MAILPOET-6416]

lysyjan committed:
Use PremiumModal component in poweredByMailpoet block

This change needed some small adjustments to in webpack config.
[MAILPOET-6416]

lysyjan committed:
Fix loading styles for custom block types

[MAILPOET-6416]

lysyjan committed:
Add styles for the premium modal and button

[MAILPOET-6416]

lysyjan committed:
Fix dark logo in the block powered-by-mailpoet

The previous logo had transparent background and the color was sent via CSS, which caused issue in the rendered email and preview.
[MAILPOET-6416]

triple0t reviewed (commented):
Awesome work, @lysyjan 

The code looks good, but I found some issues when testing.

**One**
![Screenshot 2025-01-10 at 5 36 44 PM](https://github.com/user-attachments/assets/db78f260-43c3-45f9-91ff-cfbffadcc699)
I can see the logo in the editor but not in the mail sent
![Screenshot 2025-01-10 at 5 38 38 PM](https://github.com/user-attachments/assets/44b7f506-d1e3-4fea-9378-db3af75d6037)

**Two**
![Screenshot 2025-01-10 at 5 36 25 PM](https://github.com/user-attachments/assets/3a65b577-ce21-4c2f-a23c-4e4401008245)
The logo has a weird border outline, which doesn't look nice.

**Three**
What logic determines when the logo is displayed?
The logo is displayed for me when I have a valid Premium API key and also when I have the Premium plugin installed.

lysyjan committed:
Fix property name containing active premium

The name didn't match the name in MailPoet module which is used in PoweredByMailpoet block.
[MAILPOET-6416]

lysyjan commented:
Hey @triple0t!

> I can see the logo in the editor but not in the mail sent

I have tested the sending, and the logo was displayed in the editor and in the emails that were sent. I tried sending on local and via MSS too.

> The logo has a weird border outline, which doesn't look nice.

You need to compile JS because it contains some additional CSS.

> What logic determines when the logo is displayed?
The logo is displayed for me when I have a valid Premium API key and also when I have the Premium plugin installed.

I was able to replicate it.
There was a bug with a different name configured in the EditorPageRenderer class. It is fixed in the last commit.
Nice catch 👍

triple0t commented:
Thanks for fixing some of the issues, @lysyjan 

>> I can see the logo in the editor but not in the mail sent

> I have tested the sending, and the logo was displayed in the editor and in the emails that were sent. I tried sending on local and via MSS too.

>> The logo has a weird border outline, which doesn't look nice.

> You need to compile JS because it contains some additional CSS.


I'm testing using the release zip build from circleci.

The logo is not included in the sent mails and the logo on the editor does have an outline

![Screenshot 2025-01-11 at 1 43 51 PM](https://github.com/user-attachments/assets/bab12bca-a756-4c87-a98a-88c1040e711f)

lysyjan committed:
Fix loading block.json for custom blocks

The block didn't work on production mode because the path was incorrect and the json file was not included in build.
[MAILPOET-6416]

lysyjan committed:
Remove redundant condition

After using wp_register_script and wp_register_style for custom blocks the condition is not needed anymore.
[MAILPOET-6416]

lysyjan committed:
Remove dark logo option form powered by mailpoet block

The dark logo does not have many usage with dark background a with transparent can be misused.
I decided to remove this option.
[MAILPOET-6416]

lysyjan commented:
@triple0t Thanks for catching this. The local worked fine but the production build was totally broken for the powered by mailpoet block. I found the issue and it didn't work before. I also removed the dark logo after a discussion with Rosta.
Please take a look. I want to add it to the next release.

triple0t reviewed (approved):
LGTM
